make/devices.cmake

```diff
@@ -736,19 +855,94 @@ function(add_testbench)
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     VERBATIM
     )
-  add_custom_command(
-    OUTPUT ${NAME}.vcd
+
+  # This target always just executes the testbench.  If the user wants to view
+  # waves generated from this executation, they should just build ${NAME}_view
+  # not ${NAME}.
+  add_custom_target(
+    ${NAME}
     COMMAND ${VVP} -v -N ${NAME}.vpp
     DEPENDS ${VVP} ${NAME}.vpp
     )
-  add_custom_target(
-    ${NAME} DEPENDS ${NAME}.vcd
-    )
 
   get_target_property_required(GTKWAVE env GTKWAVE)
+  add_custom_command(
+    OUTPUT ${NAME}.vcd
+    COMMAND ${VVP} -v -N ${NAME}.vpp
+    DEPENDS ${VVP} ${NAME}.vpp
+    )
   add_custom_target(
     ${NAME}_view
     DEPENDS ${NAME}.vcd
     COMMAND ${GTKWAVE} ${NAME}.vcd
     )
 endfunction()
+
+function(generate_pinmap)
+  # ~~~
+  #   GENERATE_PINMAP(
+  #     NAME <name of file to output pinmap file>
+  #     TOP <module name to generate pinmap for>
+  #     BOARD <board to generate pinmap for>
+  #     SOURCES <list of sources to load>
+  #   )
+  # ~~~
+  #
+  # Generate pinmap blindly assigns each input and output from the module
+  # ${TOP} to valid pins for the specified board. In its current version,
+  # GENERATE_PINMAP may assign IO to global wire.
+  #
+  # TODO: Consider adding knowledge of global wires and be able to assign
+  # specific wires to global wires (e.g. clock or reset lines).
```

Comment:
Maybe log an issue about this? Could probably be a good trivial "first issue" for someone -- adding support for [improving the layout csv output](https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/ice40/utils/ice40_import_layout_from_icebox.py) would be the first step I think?

Reply:
#224